cmd/parser/parse.go

```diff
@@ -0,0 +1,356 @@
+		entries = conf.Rules
+	} else {
+		entries = strings.Split(cmd.rule, ";")
+	}
+
+	allResults := make([]rules.ParsingResult, 0, len(entries))
+	okResults := make([]rules.ParsingResult, 0, len(entries))
+	errorResults := make([]rules.ParsingResult, 0, len(entries))
+
+	resolvedRules := make(map[string]rules.ParsingResult)
+
+	for _, entry := range entries {
+		rule, err := cmd.parser.Parse(entry)
+
+		result := rules.ParsingResult{OriginalRule: entry,  Rule: rule, Err: err}
+		
+		if err != nil {
+			errorResults = append(errorResults, result)	
+		} else {
+			okResults = append(okResults, result)	
+			if rule.IsResolved() {
+				resolvedRules[rule.StringNoLabels()] = result
+			}
+		}
+
+		allResults = append(allResults, result)
+	}
+
+	if cmd.sort {
+		allResults = rules.SortRuleEntries(allResults)
+		okResults = rules.SortRuleEntries(okResults)
+		errorResults = rules.SortRuleEntries(errorResults)
+	}
+
+	if cmd.unique {
+		uniqnuessSet := make(map[string]rules.ParsingResult)
+		signatureSet := make(map[string]rules.ParsingResult)
+		uniqueResults := make([]rules.ParsingResult, 0, len(allResults))
+
+
+		for _, result := range allResults {
+
+			// containsWildcards := false
+
+			if result.Err != nil {
+				uniqueResults = append(uniqueResults, result)
+				continue
+			}
+
+			negativeSignatureKey := result.Rule.Plan
+			signatureKey := result.Rule.Plan
+			key := result.Rule.Plan
+			
+			key += "PR:" 
+			signatureKey += "PR:"
+			negativeSignatureKey += "PR:"
+			if result.Rule.PlatformRegion == "" || result.Rule.PlatformRegion == "*" {
+				key += "*"
+
+				signatureKey += "*"
+				negativeSignatureKey += "attr"
+				if result.Rule.PlatformRegion == "*" {
+					// containsWildcards = true
+				}
+			} else {
+				key += result.Rule.PlatformRegion
+			
+				signatureKey += "attr"
+				negativeSignatureKey += "*"
+			}
+			
+			key += "HR:"
+			signatureKey += "HR:"
+			negativeSignatureKey += "HR:"
+			if result.Rule.HyperscalerRegion == "" || result.Rule.HyperscalerRegion == "*" {
+				key += "*"
+			
+				signatureKey += "*"
+				negativeSignatureKey += "attr"
+
+			
+				if result.Rule.HyperscalerRegion == "*" {
+					// containsWildcards = true
+				}
+
+			} else {
+				key += result.Rule.HyperscalerRegion
+			
+				signatureKey += "attr"
+				negativeSignatureKey += "*"
+			}
+
+			negativeSignatureItem, negativeSignatureExists := signatureSet[negativeSignatureKey]
+
+			// if negativeSignatureExists && containsWildcards && cmd.signature {
+			if negativeSignatureExists && cmd.signature {
+				
+				resolvingSignaturePossibleRule := result.Rule.Combine(*negativeSignatureItem.Rule)
+
+			resolvingKey := resolvingSignaturePossibleRule.StringNoLabels()
+			_, resolvingRuleExists := resolvedRules[resolvingKey]
+
+				if !resolvingRuleExists {
+					err := fmt.Errorf("Duplicated negative signature with previously defined rule: '%s', consider introducing a resolving rule '%s'", negativeSignatureItem.Rule.StringNoLabels(), resolvingKey)
```

Comment:
Negative signature is kind of cryptic for the user. We need to invent something simpler or add a section to documentation to explain it properly.